src/core/image.js

```diff
@@ -641,6 +643,35 @@ var PDFImage = (function PDFImageClosure() {
         this.undoPreblend(imgData.data, drawWidth, actualHeight);
       }
 
+      if (transferFnsIR) {
+        var numTransferFnsIR = transferFnsIR.length;
+        var numChannels = alpha01 === 0 ? 3 : 4;
+
+        if (numTransferFnsIR < numChannels) {
+          warn('Too few transfer functions.');
+          return imgData;
+        }
+        var transferFnIR, transferFn, j, jj;
+        var transferMap = new Uint8Array(256), tmp = new Float32Array(1);
```

Comment:
We shall process all three components of the RGB color at once (see http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#page=309&zoom=140,-42,721). I think transferMap cache will be huge 256x256x256 for all three components, so we need to do it without caching below. It might be slower, but right from spec point of view.

(I'm thinking not support transfer function for other CSs yet)